### PR TITLE
[Core] Adding start stop to logger

### PR DESCRIPTION
**Pull request by pooyan-dadvand**
This PR adds two new Start and Stop methods to the Logger. These methods call the Start and Stop methods of the Timer with corresponding call path so you may distinguish between i.e build and solve of different solvers.

Apart from the added methods, this PR changes the LoggerMessage::Category from an enum class to local flags. So now it is possible to tag with more than one category a message. Nevertheless, this change made it backward incompatible.

The start-stop macros for info severity are added. And a test is provided.

For example calling:

```c++
KRATOS_INFO_START("TimeStep") << time << std::endl;
KRATOS_INFO("TestLevel") << "Level 1\n";
KRATOS_INFO_START("Build");
KRATOS_INFO("TestLevel") << "Level 2\n";
KRATOS_INFO_STOP("Build");
KRATOS_INFO_START("Solve");
KRATOS_INFO("TestLevel") << "Level 2\n";
KRATOS_INFO_STOP("Solve");
KRATOS_INFO_STOP("TimeStep");
```

Would result in the following output by logger:
```
TimeStep: 0.1
  TestLevel: Level 1
  Build started
    TestLevel: Level 2
  Build finished
  Solve started
    TestLevel: Level 2
  Solve finished
TimeStep finished
```

And the resume of Timer would be:

```
                                 Repeat #               Total                   Max                     Min             Average                  Time%
000./TimeStep.......................... 1               4.449999E-05s           4.449999E-05s           4.449999E-05s    4.449999E-05s           1.363017E-01%
001./TimeStep/Build.................... 1               8.326955E-06s           8.326955E-06s           8.326955E-06s    8.326955E-06s           2.550513E-02%
002./TimeStep/Solve.................... 1               4.719011E-06s           4.719011E-06s           4.719011E-06s    4.719011E
```

### CODE REVIEW

**Comment by pooyan-dadvand** [comment]
I forgot to mention that there is an unrelated change in run_cpp_tests.py in this PR which I think is a useful feature so I left it.

File: kratos/input_output/logger.cpp [hunk 1]
```diff
@@ -57,6 +86,15 @@ namespace Kratos
     }
   }
 
+  std::string Logger::CreateFullLabel(){
+    auto& labels_stack = GetLablesStackInstance();
```
Comment (ddiezrod) [inline kratos/input_output/logger.cpp]: typo, `GetLabelsStackInstance`

**Comment by ddiezrod** [comment]
@KratosMultiphysics/technical-committee 
I see this PR is quite abandoned but I think its a very important change for us. (I do not think it makes sense to do any scalability test in MPI until this is merged.)

I have two or three doubts about it.
- Output is very verbose. Right now, every time timer is called, we print the label in the screen, this, in general, does not provide any useful information and I think it could be safely removed. (At least when there is no message behind).

- In some parts of the core (specially in Block B&S) there are some Timer::Start/Stop calls. Should we change them here?

-If we are going to stick to  Logger to do the profiling, maybe we should add a warning in the old timer.

**Comment by philbucher** [comment]
2 general thoughts 
Why is it depending on the severity? I think for your purpose it would make more sense to have it independent 
Also can you rename to sth like scope-begin instead of start? I was super confused at first, start/stop are usually related go timings 

Btw maybe it makes sense to have a separate severity for timings? I think this can be handy on several levels. They could easily be filtered and switched on and off

**Comment by philbucher** [comment]
@ddiezrod you mainly need this for timings? Then I think it would really make sense to add a timing severity I would say

**Comment by ddiezrod** [comment]
Yes @philbucher , I am only interested in timers, I dont know actually what other use this could have. In fact, arriving at this point, I would simply use the timers using the nested-labels.

**Comment by philbucher** [comment]
I though this was no longer needed with #7374?

**Comment by ddiezrod** [comment]
> I though this was no longer needed with #7374?

@pooyan-dadvand thinks this is important if we want to pass messages in timers, so I'm giving this another try. I honestly do not care too much which solution we use, I just need something asap.

**Pull request closed without merging**